host/bcm283x/gpio.go

```
@@ -462,42 +440,51 @@ func (p *Pin) DefaultPull() gpio.Pull {
 
 // Internal code.
 
+func (p *Pin) haltDMA() error {
+	if p.dmaCh != nil {
+		p.dmaCh.reset()
+		p.dmaCh = nil
+	}
+	if p.dmaBuf != nil {
+		if err := p.dmaBuf.Close(); err != nil {
+			return p.wrap(err)
+		}
+		p.dmaBuf = nil
+	}
+	return nil
+}
+
 // haltClock disables the GPCLK/PWM clock if used.
 func (p *Pin) haltClock() error {
+	p.haltDMA()
```

don't ignore the error value

---

done.